File: modul_praktikum_python/contoh_code/python/array_prosedur.py
"""Prosedur isi dan cetak array [1..3]."""


def isi(nilai: list[int]) -> None:
    i = 1
    while i <= 3:
        nilai[i] = int(input(f"Nilai ke {i} = "))
        i += 1


def cetak(nilai: list[int]) -> None:
    for i in range(1, 4):
        print(f"Nilai ke {i} = ", nilai[i])


def main() -> None:
    nilai = [0] * 4
    print("Mengisi elemen array")
    isi(nilai)
    print("Mencetak elemen array")
    cetak(nilai)


if __name__ == "__main__":
    main()

File: modul_praktikum_python/contoh_code/pascal/array_prosedur.pas
uses crt; 
type 
  Tarray=array[1..3] of integer; 

procedure isi(var nilai:Tarray); 
var 
  i:integer; 
begin 
  i:=1; 
  while (i<=3) do 
  begin  
   write(‘Nilai ke ‘,i,’ = ‘);readln(nilai[i]); 
   i:=i+1;  
  end; 
end; 

procedure cetak(nilai:Tarray); 
var 
 i:integer; 
begin 
 for i:=1 to 3 do 
 begin 
  writeln(‘Nilai ke ‘,i,’ = ‘,nilai[i]); 
 end; 
end; 

var 
 nilai:Tarray; 
 n:integer;(*data yang dicari*) 
begin 
  writeln(‘Mengisi elemen array’); 
  isi(nilai); 
  writeln(‘Mencetak elemen array’); 
  cetak(nilai); 
end.
